### DISCUSSION TITLE
How to recognize a uaObject is Event or not? And how to get the EventType of the event?

### DISCUSSION THREAD

**Discussion by simonalexs** [eclipse-milo/milo#956] [General]
`UaObjectNode objectNode = addressSpace.getObjectNode(NodeId.parse("ns=2;s=eventObject"));`

I try to browse all the RootNodes, and now I want to recognize the objectNode is Event or not.How can i do this?
Use the "TypeDefinition" of the object？Or use "NotifierOf"(one of the References)?

If use "NotifierOf", how can i get the "NotifierOf" info? I found in all source code, but didn't find usefulInfo.

**Comment by kevinherron** [comment]
Yes I think checking the Type Definition. It should already be narrowed down, though, because Object instances for events don’t appear in the general address space. EventFactory has its own NodeManager that all instances and their members are stored in until the event is deleted.

**Reply by simonalexs** [reply]
Thanks for your help.And another question I have is how to call method which takes a customDataType as argument?
I find the question in stackoverflow but it is 3 years ago, and the solution that the  answer says is too old for the eclipse-milo of the new version.Could you give me a idea to resolve this question? Thank you very much!

The question's url in stackoverflow is:
[https://stackoverflow.com/questions/53389075/eclipse-milo-how-to-call-method-which-takes-a-customdatatype-as-argument](url)

**Reply by kevinherron** [reply]
The answer is still pretty much the same - the server must support the OPC UA 1.03 DataType Dictionary mechanism, and you would need to have a `DataTypeDictionarySessionInitializer` installed like this example does https://github.com/eclipse/milo/blob/master/milo-examples/client-examples/src/main/java/org/eclipse/milo/examples/client/UnifiedAutomationReadCustomDataTypeExample.java

Then you would build a struct similar to how the Stack Overflow answer describes.

**Reply by simonalexs** [reply]
Thanks for your answer, I will try your suggestion.This helps me to indicate the direction of my thought.Thanks!
